buildSrc/src/main/groovy/org/apache/beam/gradle/VendorJavaPlugin.groovy

```diff
@@ -48,6 +49,10 @@ class VendorJavaPlugin implements Plugin<Project> {
     project.ext.vendorJava = {
```

Comment:
This is similar to how we've setup our other custom Gradle plugins (i.e. `applyXYZNature(..)` methods which take a configuration object. It differs from the convention used by most Gradle plugins, and I think the subtle difference is confusing for people just starting to learn Gradle / Groovy.

This doc shows how to set up a conventional plugin: https://docs.gradle.org/current/userguide/custom_plugins.html#sec:getting_input_from_the_build

The different would be to register the config object in `project.extensions` and drop the explicit `vendorJava()` method:

```groovy
void apply(Project project) {
  def extension = project.extensions.create('vendorJava', VendorJavaPluginConfig)
  // .. everything else from previous 'vendorJava' method inline here,
  // referencing config from 'extension' variable
}
```

and then the usage looks slightly more conventional:

```groovy
apply plugin: org.apache.beam.gradle.VendorJavaPlugin

vendorJava {
  // configuration block
}
```

Reply:
Kenn tried to get this to work in https://github.com/apache/beam/pull/6809 but was unsuccessful due to the order in which things were being evaluated since the extension only gets populated after apply(Project project) returns which meant that you couldn't get the properties out of it. The examples on the site are all about reading the extension during task execution.

Is this something that can be improved upon later?

Reply:
Bummer, I guess it's not as easy as I'd hoped. Thanks for looking into it.

Yes, we can defer this. I'd like to clean up our common Gradle code to make it easier to understand, but no need to hold this PR to a higher standard than the status quo.